ims/database/db_connection.py

```diff
@@ -1,14 +1,22 @@
 from sqlalchemy import create_engine
 from sqlalchemy.ext.declarative import declarative_base
 from sqlalchemy.orm import sessionmaker
-from sqlalchemy.pool import NullPool
 from sqlalchemy.pool import SingletonThreadPool
 
 import ims.common.config as config
+import ims.common.constants as constants
 
 _cfg = config.get()
 
 
+# ISSUE
+# This Class is not creating the engine once as previously expected
+# Although there is no harm in creating the engine and session maker multiple
+# times, it doesnt make sense when it comes to multi threading stuff as there
+# is supposed to be one engine and one session per request.
+
+# Will Probably need to build Engine Abstraction To Solve it properly
+
```

Comment:
Any specific reason for moving to this engine creation. I might have forgotten the discussion, so let me know, if this was something that was agreed upon.